datalad/__init__.py

```diff
@@ -12,6 +12,14 @@
 
 # Other imports are interspersed with lgr.debug to ease troubleshooting startup
 # delays etc.
+
+# If there is a bundled git, make sure GitPython uses it too:
+from datalad.cmd import GitRunner
+_git_runner = GitRunner()
+if _git_runner._GIT_PATH:
+    import os
+    os.environ['GIT_PYTHON_GIT_EXECUTABLE'] = os.path.join(_git_runner._GIT_PATH, 'git')
```

Comment:
Re start time: I think that is ill-measured. As far as I'm aware, there is no datalad cmdline tool, that wouldn't import `gitrepo.py` anyway. Even `datalad --help` needs to import all the toplevel functions, which in turn are importing `gitrepo` directly or via `annexrepo` or via `dataset` ... I don't think there would be an actual difference, the user could possibly experience. I think this is only valid for an artificial measure point with no meaning regarding the actual use of any command.

Re test: Well, but what would it actually test? At what point the setting should be in effect according to that test? I'm not opposing a proper test, I'm just not sure, what's actually appropriate.

Re using protected API: Well, okay. Will do it that way.